### PR TITLE
Chunk and upload files greater than 4kb

### LINKED ISSUE #81: Upload file splitting
Files can not be larger than 4kB.  If upload fails with some error which indirectly means: "file too big" (upstream... connection error...), chop file in half and try again with create for first half and fileAppendTransaction for second half - recursively do these halves until no error.

### PR DESCRIPTION
**Pull request by QuestofIranon**
closes #81

### CODE REVIEW

File: src/components/UploadFile.vue [hunk 1]
```diff
@@ -150,30 +152,70 @@ export default createComponent({
...
 
+                // TODO[2019-10-18]: handle this properly lol
+                if (!fileId) {
+                    throw new Error("no file id :(");
+                }
+
+                for (const chunk in chunks) {
+                    console.log("chunk!");
+
+                    await new FileAppendTransaction(client as InstanceType<
```
Comment (Sheng-Long) [inline src/components/UploadFile.vue]: Trying .svg file @ 25.9 kB and it errors out with `Error: "Hedera returned response code: FILE_CONTENT_EMPTY (47)"`.

File: src/components/UploadFile.vue [hunk 2]
```diff
@@ -150,30 +152,70 @@ export default createComponent({
...
                     .executeForReceipt();
 
                 const fileId = receipt.value.fileId;
 
+                // TODO[2019-10-18]: handle this properly lol
+                if (!fileId) {
+                    throw new Error("no file id :(");
+                }
+
+                for (const chunk in chunks) {
```
Comment (Sheng-Long) [inline src/components/UploadFile.vue]: Seems chunk is of type string here

File: src/components/UploadFile.vue [hunk 3]
```diff
@@ -150,30 +152,70 @@ export default createComponent({
...
+                // TODO[2019-10-18]: send alert to user and back out of interface
+                if (!publicKey) {
+                    throw new Error("not logged in");
+                }
+
                 receipt.value = await new FileCreateTransaction(
                     client as InstanceType<typeof Client>
                 )
-                    .setContents(state.fileUint8Array as Uint8Array)
+                    .setContents(chunks.pop() as Uint8Array)
```
[the remark this answers is not in the document]
Comment (QuestofIranon) [reply src/components/UploadFile.vue]: Yeah I was treating it like a queue where `pop` would pull the first (oldest) part of the file.

File: src/components/UploadFile.vue [hunk 4]
```diff
@@ -150,30 +152,70 @@ export default createComponent({
...
+                    const start = i * MAX_FILE_LENGTH;
+                    chunks.push(file.slice(start, start + MAX_FILE_LENGTH));
+                }
 
                 const receipt = ref<TransactionReceipt | null>(null);
 
+                const publicKey = (await store.state.wallet.session.wallet.getPublicKey()) as import("@hashgraph/sdk").Ed25519PublicKey;
+
+                // TODO[2019-10-18]: send alert to user and back out of interface
+                if (!publicKey) {
```
[the remark this answers is not in the document]
Comment (QuestofIranon) [reply src/components/UploadFile.vue]: `null` counts as `false` in boolean contexts due to falsiness.
https://developer.mozilla.org/en-US/docs/Glossary/Falsy

**Pull request merged**